added two remove features for P4J

SophieHYe opened the pull request:
Hi @martinezmatias,

In this PR, I propose two code removal related features: Remove_Guard and Remove_Statement.
Previous, the code deletion was considered Insert_Guard.  The two new test cases are also added to this PR.

martinezmatias commented on src/main/java/fr/inria/prophet4j/feature/original/OriginalRepairGenerator.java:
```diff
@@ -296,7 +296,10 @@ public Repair obtainHumanRepair() {
                 // GuardKind: // INSERT_GUARD_RF
-                repair.kind = RepairKind.GuardKind;
+                repair.kind = RepairKind.RemoveSTMTKind;
+                if (diffEntry.srcNode instanceof CtIf) {
```
Hi @SophieHYe 
I wonder whether we should add an additional check in this case.
Currently, it checks if a removed element is an IF, which is fine.
However, this case would also  mark as "positive" diffs that also removes the If's blocks such as:
```
- if(a>0){
-      a = b;
- }
```

In that case, I am not sure if the feature REMOVE_GUARD_RF aims at capturing that case.
I would imagine that it focus on removes of If but its block(s) then/else is not removed
```
- if(a>0){
      a = b;
-}
```
WDYT?
If we only focus on  detecting removes of the If (and not the then block), it would be necessary  to check if the Block was removed (i.e., if there is a `not root` action Delete which node is the element referenced by the If's then) or was not removed (i.e., if there is a Move action which node is is the element referenced by the If's then).

SophieHYe replied:
Hi @martinezmatias, that's good point. I was only considering two remove types either they are in if block or a normal statement. I will see if I can update it to remove_whole_block or remove_only_guard or something others. I will update this PR!

martinezmatias replied:
Hi @SophieHYe 
Perfect!
> I was only considering two remove types either they are in if block or a normal statement
Okey. One possibility is to simply rename the feature you create `REMOVE_GUARD_RF` by something like `REMOVE_IF_BLOCK` with the goal of avoiding confusion with the `guard` terminology.

martinezmatias commented:
Hi @SophieHYe 
Thanks a lot for the PR.
I wrote one comment about one doubt I have about one of those features.

SophieHYe commented:
Hi @martinezmatias,
I updated this PR wrt your suggestions. Currently, the names of the features: REMOVE_STMT, REMOVE_PARTIAL_IF, REMOVE_WHOLE_IF, REMOVE_WHOLE_BLOCK(rather than if block).

REMOVE_WHOLE_IF:
```
- if(a>0){
-      a = b;
-}
```
REMOVE_PARTIAL_IF:
```
- if(a>0){
      a = b;
-}
```
I was working on REMOVE_PARTIAL_IF, I fixed the bug to get DiffEntry. Due to the REMOVE_PARTIAL_IF contains two operations: delete and move, the original buggy version considers the last operation in the source nodes, in this case, which leads to the node of the MOV operation instead of the DEL node. So I conclude the operation to delete and partially delete(delete + move). The case cases have been updated too.
I will continue to check if other operations (insert and update) are correct.

martinezmatias commented:
Hi @SophieHYe 
Thanks for the update of the PR.
It looks fine for me.
Merged
